packages/app/contexts/ModalContext.tsx

```diff
@@ -0,0 +1,48 @@
+"use client";
+
+import { createContext, useContext, ReactNode, useState, useMemo } from "react";
+
+export enum ModalId {
+  CONFIRM_STACK = "confirmStack",
+  STACK = "stack",
+  TOKEN_PICKER = "tokenPicker",
+}
+
+export interface ModalContextProps {
+  modalToOpen: ModalId | null;
+  closeModal: () => void;
+  openModal: (id: ModalId) => void;
+}
+
+export const ModalContext = createContext<ModalContextProps>({
+  modalToOpen: null,
+  closeModal: () => {},
+  openModal: (id: ModalId) => {},
+});
+
+export interface ModalContextProviderProps {
+  children: ReactNode;
+}
+
+export const ModalContextProvider = ({
+  children,
+}: ModalContextProviderProps) => {
+  const [modalToOpen, setModalToOpen] = useState<ModalId | null>(null);
```

Comment:
related to comment above:

```suggestion
  const [modalToOpen, setModalOpenId] = useState<ModalId | null>(null);
```